mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraBlobsDAOTest.java

```diff
@@ -0,0 +1,126 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.apache.commons.io.Charsets;
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.cassandra.BlobId;
+import org.apache.james.mailbox.cassandra.modules.CassandraBlobModule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.Strings;
+
+public class CassandraBlobsDAOTest {
+    private static final int MULTIPLE_CHUNK_SIZE = 3 * CassandraBlobsDAO.CHUNK_SIZE;
+    private CassandraCluster cassandra;
+    private CassandraBlobsDAO testee;
+
+    @Before
+    public void setUp() throws Exception {
+        cassandra = CassandraCluster.create(new CassandraBlobModule());
+        cassandra.ensureAllTables();
+
+        testee = new CassandraBlobsDAO(cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cassandra.clearAllTables();
+        cassandra.close();
+    }
+
+    @Test
+    public void saveShouldReturnEmptyWhenNullData() throws Exception {
+        Optional<BlobId> blobId = testee.save(null).join();
+
+        assertThat(blobId.isPresent()).isFalse();
+    }
+
+    @Test
+    public void saveShouldSaveEmptyData() throws Exception {
+        Optional<BlobId> blobId = testee.save(new byte[]{}).join();
+
+        byte[] bytes = testee.read(blobId.get()).join();
+
+        assertThat(blobId.isPresent()).isTrue();
+        assertThat(new String(bytes, Charsets.UTF_8)).isEmpty();
+    }
+
+    @Test
+    public void saveShouldSaveBlankData() throws Exception {
+        Optional<BlobId> blobId = testee.save("".getBytes(Charsets.UTF_8)).join();
+
+        byte[] bytes = testee.read(blobId.get()).join();
+
+        assertThat(blobId.isPresent()).isTrue();
+        assertThat(new String(bytes, Charsets.UTF_8)).isEmpty();
+    }
+
+    @Test
+    public void saveShouldReturnBlobId() throws Exception {
+        Optional<BlobId> blobId = testee.save("toto".getBytes(Charsets.UTF_8)).join();
+
+        assertThat(blobId.isPresent()).isTrue();
+    }
+
+    @Test
+    public void readShouldBeEmptyWhenNoExisting() throws IOException {
+        byte[] bytes = testee.read(BlobId.from("unknown")).join();
+
+        assertThat(bytes).isEmpty();
+    }
+
+    @Test
+    public void readShouldReturnSavedData() throws IOException {
+        Optional<BlobId> blobId = testee.save("toto".getBytes(Charsets.UTF_8)).join();
+
+        byte[] bytes = testee.read(blobId.get()).join();
+
+        assertThat(new String(bytes, Charsets.UTF_8)).isEqualTo("toto");
+    }
+
+    @Test
+    public void readShouldReturnLongSavedData() throws IOException {
+        String longString = Strings.repeat("0123456789\n", 1000);
```

Comment:
I would replace by 1000 by `MULTIPLE_CHUNK_SIZE + 1` to make clear and to be sure it's not a perfect multiple of `CHUNK_SIZE`